curriculum/js1/9.js

```diff
@@ -5,7 +5,10 @@
  */
 
 const solution = (a) => {
-  return 0;
+  if ((a % 7) === 0) {
+    return a+7;
+  }
+  return solution (a-1);
```

Comment:
why are you subtracting 1?

Reply:
if input is 6, your solution will return 8. But it should return 7

Reply:
```javascript
const solution = (a) => {
  if ((a % 7) === 0) {
    return a+7;
  }
  return solution (a-1);
};

solution(6);
// 6 % 7 === 6, so we call solution(5)
// 5 % 7 === 6, so we call solution(4)
// 4 % 7 === 6, so we call solution(4)
// ...
// 1 % 7 === 1, so we call solution(4)
// 0 % 7 === 0, so we return 0+7, which is the next multiple of 7 after 6
```

Reply:
Daniel basically explained it, but had a few typo's so here it is again below:

const solution = (a) => {
  if ((a % 7) === 0) {
    return a+7;
  }
  return solution (a-1);
};

solution(6);
// 6 % 7 === 6, so we call solution(5)
// 5 % 7 === 6, so we call solution(4)
// 4 % 7 === 6, so we call solution(3)
// ...
// 1 % 7 === 1, so we call solution(0)
// 0 % 7 === 0, so we return 0+7, which is the next multiple of 7 after 6

Reply:
Did you have a more efficient solution? please let me know!

Reply:
no, this is a brilliant solution. Thanks for sharing.